Vikta-API-testing-sample/src/test/java/com/test/api/imageitem/ImageItemTest.java

```diff
@@ -0,0 +1,197 @@
+
+        assertThat(response.getStatusCode()).isEqualTo(404);
+    }
+
+    @Test
+    public void canListAllImageItems(){
+        List<ImageItemDTO> imageItemsList = getImageItemsList();
+        Set<Long> categoryIds = new HashSet<>();
+        List<String> tags = new ArrayList<>();
+        Double price = 1.0;
+
+        ImageItemDTO emptyImageItem = new ImageItemDTO();
+        emptyImageItem.setPathToImage("");
+        emptyImageItem.setTitle("");
+        emptyImageItem.setCategoryIds(categoryIds);
+        emptyImageItem.setTags(tags);
+        emptyImageItem.setAuthor("");
+        emptyImageItem.setDescription("");
+        emptyImageItem.setPrice(price);
+        emptyImageItem.setRating(0);
+
+        assertThat(imageItemsList).as("ImageItem list shouldn't be empty").isNotEmpty();
+
+        assertThat(imageItemsList).as("ImageItem fields shouldn't be empty")
+                .allSatisfy(imageItem -> assertThat(imageItem)
+                        .usingRecursiveComparison().ignoringFields("description", "price", "rating")
+                        .isNotEqualTo(emptyImageItem));
+
+        assertThat(imageItemsList).as("Rating should be in range [0,5]")
+                .allSatisfy(imageItem -> assertThat(imageItem.getRating())
+                        .isBetween(0,5));
+
+        assertThat(imageItemsList).as("Price shouldn't be negative")
+                .allSatisfy(imageItem -> assertThat(imageItem.getPrice())
+                        .isPositive());
+    }
+
+    @Test
+    public void canSearchImageItemByTerm() {
+        final String termQueryParam = "term";
+        Random random = new Random();
+        List<ImageItemDTO> imageItemsList = getImageItemsList();
+
+        assertThat(imageItemsList)
+                .as("ImageItem list is empty, can't check if can find ImageItem")
+                .isNotEmpty();
+
+        int size = random.nextInt(imageItemsList.size()) + 1;
+
+        String term = imageItemsList.stream()
+                .map(ImageItemDTO::getTags)
+                .limit(size)
+                .map(imageItem -> imageItem.toString()
+                        .substring(1,imageItem.size())
+                        .replaceAll(" ", "~"))
+                .collect(Collectors.joining("|"));
+
+        List<ImageItemDTO> resultList = given().spec(defaultRequestSpec())
+                .queryParam(termQueryParam, term)
+                .when()
+                .log().all()
+                .get(getTestEnvironment().getImageitemSearchPath())
+                .then()
+                .log().all()
+                .spec(defaultResponseSpec())
+                .extract().body().jsonPath().getList(".", ImageItemDTO.class);
+
+        assertThat(resultList)
+                .as("ImageItem search result list shouldn't be empty")
+                .isNotEmpty();
+
+        assertThat(resultList.size())
+                .as("Should find more imageItems")
+                .isGreaterThan(size);
```

Comment:
to correct - list don't have to be greater, because when one imageitem has two tags form term, then won't be returned twice